### PR TITLE
Add the minesweeper exercise and test

### PR DESCRIPTION
**Pull request by arthurchipdean**

### CODE REVIEW

**Comment by rchavarria** [comment]
Great! And thanks for adding a new exercise to the track.

In the `config.json` file, you added the `minesweeper` exercise in the last position. That item you added to the `exercises` array has three properties: `slug`, `difficulty` and `topics`.

You valued it with a `difficulty` of 7. It'd be really cool if you could fill the `topics` property too. A list of available topics could be found here: https://github.com/exercism/x-common/blob/master/TOPICS.txt

**Comment by matthewmorgan** [comment]
@arthurchipdean thanks a lot for doing this.  I have some requests on small style changes to the code, just to keep things consistent with the rest of the track.  I'll call them out in specific line comments.

File: exercises/minesweeper/example.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  for(var x = 0; x < board.length; x++) {
+    outBoard[x] = [];
+    for(var y = 0; y < board[x].length; y++) {
+      var spot = board[x][y];
+      if(spot == "*") {
+        outBoard[x][y] = spot;
+        continue;
+      }
+      var bombCount = this.distanceXdistanceYs.map(function(dxdy) {
+        if(board[x + dxdy[0]] === undefined) return 0;
```
Comment (matthewmorgan) [inline exercises/minesweeper/example.js]: Let's use curly braces here, even though they're not strictly required.  

And a space between `if` and `(` please.

File: exercises/minesweeper/example.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  if(rows.length === 0 || rows[0].length === 0) {
+    return rows;
+  }
+  var board = rows.map(function(row) { return row.split(""); });
+  var outBoard = [];
+  for(var x = 0; x < board.length; x++) {
+    outBoard[x] = [];
+    for(var y = 0; y < board[x].length; y++) {
+      var spot = board[x][y];
+      if(spot == "*") {
```
Comment (matthewmorgan) [inline exercises/minesweeper/example.js]: `===` please :)

File: exercises/minesweeper/minesweeper.spec.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        "***"
+      ];
+    var expected = [
+        "***",
+        "*8*",
+        "***"
+      ];
+    expect(minesweeper.annotate(input)).toEqual(expected);
+  });
+  xit('handles horizontal line', function() {
```
Comment (matthewmorgan) [inline exercises/minesweeper/minesweeper.spec.js]: A blank line here would be nice, and consistent with the rest of your lovely tests.

File: exercises/minesweeper/example.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+Minesweeper.prototype.annotate = function(rows) {
+  if(rows.length === 0 || rows[0].length === 0) {
+    return rows;
+  }
+  var board = rows.map(function(row) { return row.split(""); });
+  var outBoard = [];
+  for(var x = 0; x < board.length; x++) {
+    outBoard[x] = [];
+    for(var y = 0; y < board[x].length; y++) {
```
Comment (matthewmorgan) [inline exercises/minesweeper/example.js]: I think a space between `for` and `(` would be conventional.  Both lines 24 and 26.

**Comment by arthurchipdean** [comment]
I've made all the changes. Please let me know if there is anything else I should do.

**Comment by matthewmorgan** [comment]
💯 Thanks @arthurchipdean !

**Pull request merged**